spec/policies/assignment_policy_spec.rb

```diff
@@ -0,0 +1,110 @@
+describe AssignmentPolicy do
+  include PolicyHelper
+
+  describe '#run_tests?' do
+
+    context 'when the user is an admin' do
+      context 'if enable_test is false' do
+        it do
+          user = build(:admin)
```

Comment:
While I think the overall structure and choice to use local variables is totally fine, I think that the setting of `user` is static and independent enough that it should be moved into a `let` at the top of each top-level context block.

Reply:
I may be wrong, but with the custom matcher approach and without the stubbing, the let syntax for everything may be feasible again. Let me do a quick test.